assets/js/modules/analytics/common/account-create.js

```diff
@@ -0,0 +1,195 @@
+	/*
+	const isDoingCreateAccount = useSelect(
+		( select ) => {
+			return select( STORE_NAME ).isDoingCreateAccount();
+		},
+		[]
+	);
+	const accountTicketTermsOfServiceURL = useSelect(
+		( select ) => {
+			return select( STORE_NAME ).getAccountTicketTermsOfServiceURL();
+		},
+		[]
+	);
+
+	*/
+	//const { createAccount } = useDispatch( STORE_NAME );
+
+	/*
+	// Redirect if the accountTicketTermsOfServiceURL is set.
+	if ( accountTicketTermsOfServiceURL ) {
+		location = accountTicketTermsOfServiceURL;
+	}
+	*/
+
+	const isDoingCreateAccount = false;
+	const { createAccount } = () => {};
+	const { siteName, siteURL, timezone: tz } = global.googlesitekit.admin;
+	const errorCode = getQueryArg( location.href, 'error_code' );
+
+	// Handle expected provisioning flow error codes.
+	let errorMessage = false;
+	switch ( errorCode ) {
+		case 'user_cancel':
+			errorMessage = __( 'The Terms of Service were not accepted.', 'google-site-kit' );
+			break;
+
+		case 'max_accounts_reached':
+			errorMessage = __( 'The Google Analytics account limit has been reached.', 'google-site-kit' );
+			break;
+
+		case 'backend_error':
+			errorMessage = __( 'Unknown service error.', 'google-site-kit' );
+			break;
+	}
+	const [ error, setError ] = useState( errorMessage );
+
+	const handleSubmit = useCallback( ( accountName, propertyName, profileName, timezone ) => {
+		trackEvent( 'analytics_setup', 'new_account_setup_clicked' );
+		createAccount( {
+			accountName,
+			propertyName,
+			profileName,
+			timezone,
+		} ).then( ( e ) => {
+			const { error: err } = e.payload;
+			if ( err ) {
+				setError( err.message ? err.message : __( 'Unknown error.', 'google-site-kit' ) );
+			}
+		} );
+	} );
+
+	const [ accountName, setAccountName ] = useState( siteName );
+	const [ propertyName, setPropertyName ] = useState( siteURL );
+	const [ profileName, setProfileName ] = useState( __( 'All website traffic', 'google-site-kit' ) );
+	const [ timezone, setTimezone ] = useState( tz );
+	const [ validationIssues, setValidationIssues ] = useState( {
+		accountName: accountName === '',
+		propertyName: propertyName === '',
+		profileName: profileName === '',
+		timezone: timezone === '',
+	} );
+
+	// Disable the submit button if there are validation errors, and while submission is in progress.
+	const buttonDisabled = validationIssues.accountName || validationIssues.propertyName || validationIssues.profileName || validationIssues.timezone || isDoingCreateAccount;
+
+	return (
+		<Fragment>
+			<div className="googlesitekit-setup-module">
+				<div className="mdc-layout-grid__inner">
+					<div className="mdc-layout-grid__cell--span-12">
+						<h2>
+							{ __( 'Create new Analytics account', 'google-site-kit' ) }
+						</h2>
+						{
+							error &&
+							<div className="error">
+								<p>
+									{ error }
+								</p>
+							</div>
+						}
+						<div className="mdc-layout-grid">
+							{
+								isDoingCreateAccount
+									? <ProgressBar />
```

Comment:
This creates unnecessary indentation. Let's use an early return instead, further above `if ( isDoingCreateAccount ) { return <ProgressBar />; }`.

Also let's add `useState` for an `[ isNavigating, setIsNavigating ]` as we discussed the other day, to keep the loading state going until redirected to GA TOS.

Reply:
Added...  part in https://github.com/google/site-kit-wp/pull/1389/commits/d5ff025785f3a7cc1a12566392b835c6c31d5f01.